### PR TITLE
MG-888 - Add domians SDK tests

### PR DESCRIPTION
**Pull request by felixgateru**
# What type of PR is this?
This is a featurea because it adds domains SDK tests.

## What does this do?

This pr:

- Adds domains sdk tests `pkg/sdk/go/domains_test.go`
- refactors the validation error returned for empty userIds and relation in `auth/api/http/domains/requests.go`

## Which issue(s) does this PR fix/relate to?

- Related Issue #888

## Have you included tests for your changes?
Yes, the changes are to the tests.

## Did you document any new/modified feature?
No,

### CODE REVIEW

File: pkg/sdk/go/domains.go [hunk 1]
```diff
@@ -56,6 +57,9 @@ func (sdk mgSDK) UpdateDomain(domain Domain, token string) (Domain, errors.SDKEr
 		return Domain{}, errors.NewSDKError(err)
 	}
 
+	if domain.ID == "" {
+		return Domain{}, errors.NewSDKError(apiutil.ErrMissingID)
+	}
```
Comment (arvindh123) [inline pkg/sdk/go/domains.go:57]: Is there any other way instead of using apiutils?

**Review by arvindh123 — CHANGES_REQUESTED** [review]
I would like to know why `ListDomainUsers` was not covered in Test.

File: pkg/sdk/go/domains.go (hunk 1 above)
Comment (felixgateru) [reply pkg/sdk/go/domains.go:57]: I can define a new sdk error with status if needed.

Comment (arvindh123) [reply pkg/sdk/go/domains.go:57]: Lets go with now, later we can move errors if needed.

**Review by arvindh123 — DISMISSED** [review]

File: pkg/sdk/go/domains_test.go [hunk 2]
```diff
@@ -0,0 +1,1151 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package sdk_test
+
+import (
```
Comment (rodneyosodo) [inline pkg/sdk/go/domains_test.go:6]: Add test for `ListDomainUsers`

File: pkg/sdk/go/domains.go [hunk 3]
```diff
@@ -56,6 +57,9 @@ func (sdk mgSDK) UpdateDomain(domain Domain, token string) (Domain, errors.SDKEr
 		return Domain{}, errors.NewSDKError(err)
 	}
 
+	if domain.ID == "" {
```
Comment (rodneyosodo) [inline pkg/sdk/go/domains.go:55]: move the check up

File: pkg/sdk/go/domains.go [hunk 4]
```diff
@@ -9,6 +9,7 @@ import (
 	"net/http"
 	"time"
 
+	"github.com/absmach/magistrala/internal/apiutil"
```
Comment (rodneyosodo) [inline pkg/sdk/go/domains.go]: Add id check for `DomainPermissions`, `ListDomainUsers`, `ListDomainUsers`, `changeDomainStatus`, `changeDomainStatus`, `RemoveUserFromDomain`

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: pkg/sdk/go/domains_test.go (hunk 2 above)
Comment (felixgateru) [reply pkg/sdk/go/domains_test.go:6]: Covered in users tests, is refactored by #2195

File: pkg/sdk/go/domains.go (hunk 4 above)
Comment (felixgateru) [reply pkg/sdk/go/domains.go]: this check is already handled in the api level for these functions.

**Review by arvindh123 — DISMISSED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**